docs: add SETUP.MD (#217)
UiPath/uipath-mcp-python: SETUP.MD

# SETUP.MD

## Prerequisites

- Python 3.11+
- [uv](https://docs.astral.sh/uv/) 0.5+

### Supported platforms

> This checklist controls the Agentic Inner Loop KPI pipeline targets (clean Linux VM). The repo's own GitHub Actions CI additionally tests on `windows-latest`, but that is not measured by this KPI.

Reviewer: please remove the KPI references. this is a public repo

Author: Removed in commit 46ccd2e — the paragraph under `### Supported platforms` no longer mentions KPI/MER. PR title and description also reworded.

- [x] Linux
- [ ] Windows

> - [ ] macOS

Reviewer: uv is shell agnostic, it can be ran on any of the 3 platform mentioned above. the

Author: Fixed in commit 46ccd2e — all three platforms now checked (`Linux`, `Windows`, `macOS`).

## Environment Variables

### Standard (injected by pipeline)

None of the standard pipeline variables are required for environment setup, build, or unit tests.

### Project-specific

None. The unit-test suite under the `Test` section below runs fully offline and requires no external authentication.

## Setup

```bash
# Install uv if not already on PATH (no-op when pre-installed by the pipeline)
python3 -m pip install --upgrade uv

# Sync the package with dev dependencies
uv sync --all-extras
```

## Verify Setup

```bash
python3 --version
uv --version
uv run python -c "import uipath_mcp; print('uipath_mcp ok')"
```

## Build

```bash

> uv build

Reviewer: this is creating build artefacts in `dist` subdir. https://docs.astral.sh/uv/guides/package/#building-your-package
it is an intermediary step needed to publish the project to a registry. not needed as a setup step

# SETUP.MD

## Prerequisites

- Python 3.11+
- [uv](https://docs.astral.sh/uv/) 0.5+

### Supported platforms

This checklist controls the Agentic Inner Loop KPI pipeline targets (clean Linux VM). The repo's own GitHub Actions CI additionally tests on `windows-latest`, but that is not measured by this KPI.

- [x] Linux
- [ ] Windows
- [ ] macOS

## Environment Variables

### Standard (injected by pipeline)

None of the standard pipeline variables are required for environment setup, build, or unit tests.

### Project-specific

None. The unit-test suite under the `Test` section below runs fully offline and requires no external authentication.

## Setup

```bash
# Install uv if not present
curl -LsSf https://astral.sh/uv/install.sh | sh
export PATH="$HOME/.local/bin:$PATH"

# Sync the package with dev dependencies

> uv --directory . sync --all-extras

Author: Addressed in latest commit (force-pushed) — switched to `python3 -m pip install --upgrade uv` and dropped `uv --directory .` in favor of plain `uv ...` commands run from the repo root.

```

## Verify Setup

```bash
python3 --version
uv --version

> uv --directory . run python -c "import uipath_mcp; print('uipath_mcp ok')"

Author: Addressed in latest commit (force-pushed) — switched to `python3 -m pip install --upgrade uv` and dropped `uv --directory .` in favor of plain `uv ...` commands run from the repo root.

```

## Build

```bash

> uv --directory . build

Author: Addressed in latest commit (force-pushed) — switched to `python3 -m pip install --upgrade uv` and dropped `uv --directory .` in favor of plain `uv ...` commands run from the repo root.

```

## Test

```bash

> uv --directory . run pytest

Author: Addressed in latest commit (force-pushed) — switched to `python3 -m pip install --upgrade uv` and dropped `uv --directory .` in favor of plain `uv ...` commands run from the repo root.

```python
@property
def server_count(self) -> int:
    """Number of MCP servers currently loaded from configuration."""
    return len(self._servers)
```

Then create `tests/test_config_server_count.py` with two pytest tests:

```python
"""Tests for McpConfig.server_count."""

import json
from pathlib import Path

from uipath_mcp._cli._utils._config import McpConfig

def test_server_count_empty(tmp_path: Path) -> None:
    config_path = tmp_path / "mcp.json"
    config_path.write_text(json.dumps({"servers": {}}))
    cfg = McpConfig(str(config_path))
    assert cfg.server_count == 0

def test_server_count_multiple(tmp_path: Path) -> None:
    config_path = tmp_path / "mcp.json"
    config_path.write_text(
        json.dumps(
            {
                "servers": {
                    "alpha": {"command": "node", "args": ["alpha.js"]},
                    "beta": {"command": "node", "args": ["beta.js"]},
                }
            }
        )
    )
    cfg = McpConfig(str(config_path))
    assert cfg.server_count == 2
```

### Verification

```bash

> uv --directory . run pytest tests/test_config_server_count.py -v

Author: Addressed in latest commit (force-pushed) — switched to `python3 -m pip install --upgrade uv` and dropped `uv --directory .` in favor of plain `uv ...` commands run from the repo root.

# SETUP.MD

This file documents how to provision a clean development environment for `uipath-mcp`, run the build, execute the tests, and validate a sample code change end-to-end. It is intended both as a quick reference for human contributors and as a structured guide for automated environment-setup tooling.

## Prerequisites

- Python 3.11+
- [uv](https://docs.astral.sh/uv/) 0.5+

### Supported platforms

`uv` is shell- and OS-agnostic, so the commands below run unchanged on every supported platform:

- [x] Linux
- [x] Windows
- [x] macOS

## Environment Variables

### Standard

> None of the common shared environment variables (e.g. `AZURE_DEVOPS_PAT`, `GH_NPM_REGISTRY_TOKEN`) are required for environment setup, build, or unit tests.

Reviewer: why mention the env var names at all?

Author: Trimmed in latest commit — `Standard` now reads just `None required for environment setup, build, or unit tests.`

### Project-specific

None. The unit-test suite under the `Test` section below runs fully offline and requires no external authentication.

> > **All commands below must be run from the repository root.** The `uv` invocations resolve `pyproject.toml`, `src/`, and `tests/` relative to the current working directory. The first line of `## Setup` enforces this by `cd`-ing to the git root, so the rest of the file works regardless of where the user started.

Reviewer: ```suggestion
> **All commands below must be run from the repository root.** The `uv` invocations resolve `pyproject.toml`, `src/`, and `tests/` relative to the current working directory. The first line of `## Setup` enforces this by `cd`-ing to the git root.
```

Author: Applied your suggestion verbatim.

## Setup

```bash
# Ensure the working directory is the repo root (no-op if already there)
cd "$(git rev-parse --show-toplevel)"

# Install uv if not already on PATH
python3 -m pip install --upgrade uv

# Sync the package with dev dependencies
uv sync --all-extras
```

## Verify Setup

```bash
uv --version

> uv run python --version   # Reports the interpreter uv will actually use (managed by uv, not the system Python)

Reviewer: unnecessary comment

Author: Removed the trailing comment on the python version line and stripped the other self-explanatory bash comments in the Setup block.

uv run python -c "import uipath_mcp; print('uipath_mcp ok')"
```

## Build

```bash
# Compile every Python module to bytecode — fast syntax/import validation,
# no publish artifacts produced.

> uv run python -m compileall src/

Reviewer: why is this needed?

```python
@property
def server_count(self) -> int:
    """Number of MCP servers currently loaded from configuration."""
    return len(self._servers)
```

Then create `tests/test_config_server_count.py` with two pytest tests:

```python
"""Tests for McpConfig.server_count."""

import json
from pathlib import Path

from uipath_mcp._cli._utils._config import McpConfig

def test_server_count_empty(tmp_path: Path) -> None:
    config_path = tmp_path / "mcp.json"
    config_path.write_text(json.dumps({"servers": {}}))
    cfg = McpConfig(str(config_path))
    assert cfg.server_count == 0

def test_server_count_multiple(tmp_path: Path) -> None:
    config_path = tmp_path / "mcp.json"
    config_path.write_text(
        json.dumps(
            {
                "servers": {
                    "alpha": {"command": "node", "args": ["alpha.js"]},
                    "beta": {"command": "node", "args": ["beta.js"]},
                }
            }
        )
    )
    cfg = McpConfig(str(config_path))
    assert cfg.server_count == 2
```

### Verification

```bash
uv run pytest tests/test_config_server_count.py -v
```

> ## Test with an agent (coded MCP for this example)

Reviewer: ```suggestion
## Test with a real UiPath Coded MCP
```

Author: Applied your suggestion.

The unit tests above are necessary but not sufficient — they don't exercise the package end-to-end through a real agent. The flow below validates changes against a live MCP/agent runtime:

1. Apply the code changes locally.
2. Run the unit tests (see the `Sample Code Change` section above).

> 3. Scaffold a coded agent (or coded MCP) that exercises the changed code path.

Reviewer: ```suggestion
3. Scaffold a coded UiPath MCP that exercises the changed code path
```
my PR comment was just informative, you don t need to use the exact phrasing. why even mention coded agents here?

Author: Applied your suggestion — dropped the "or coded agent" qualifier and simplified to just "coded UiPath MCP" throughout the section since this repo is specifically about MCPs.
